### ISSUE TITLE
Add a version selector widget

### ISSUE THREAD

**Issue by agoose77** [jupyter-book/myst-theme#430]
At SciPy 2024, there were users who wanted a version selector to move between deployed versions of a static site (such as @mwcraig). We could adopt the pydata-sphinx-theme approach to add support for this feature as a simple `<a href>` dropdown.

This also might relate to https://github.com/readthedocs/readthedocs.org/issues/11474#issue-2410813215, which amongst other things drops the flyout HTML injection and [exposes version information through a custom event](https://github.com/readthedocs/sphinx_rtd_theme/blob/7f59a93144d0b414af9854f52811db0bded6a99f/sphinx_rtd_theme/static/js/versions.js_t#L62)

To satisfy both cases, we might want to be smart and bundle RTD + explicit versioning in a single theme plugin.

**Comment by mwcraig** [comment]
Thanks for your help getting this going for me at SciPy 😀

**Comment by humitos** [comment]
I implemented the version/language selectors for the PyData Sphinx theme as an example. You can take a look at it in this comment: https://github.com/pydata/pydata-sphinx-theme/issues/705#issuecomment-2353096494

**Comment by coretl** [comment]
We currently use sphinx + myst-parser + pydata-sphinx-theme using their JSON switcher. We would like to switch across to mystmd, but we have many repos with versioned documentation (and the JSON switcher to switch between them), and we would need the same functionality in mystmd to do this. I'd like to volunteer my team's effort to port this across, but we'd need some information to get started:
- Would this be a theme specific addition to the book theme, or a general plugin?
- Would it live in this repository or in a different repository?
- In either case, is there an example you can point to for a similar component that we can use to learn from, or point us to relevant docs?
- Would you be willing to accept a partial solution that just worked with the existing pydata JSON switcher format if it was extendable in the future to support RTD?

**Comment by choldgraf** [comment]
Something tells me we could get away with doing this via a myst plugin. I think it would look something like this:

- Load it by listing it in `plugins:` in `myst.yml`
- Call it with a role or directive like `{version-dropdown}`
- Either configure it in-line with the directive, or parse myst.yml for the configuration (would be much easier when https://github.com/jupyter-book/mystmd/issues/2949 is implemented). We could probably pull from the pydata dropdown widget for inspiration on configuration and options.
- Uses a the anywidget interface to export a dropdown at the location of the directive/role/etc. It could read the current page URL and populate the dropdown with the necessary re-write links and versions etc

Might be worth giving that a shot

**Comment by coretl** [comment]
Thanks, We'll have a go at writing a plugin and see where we get stuck. I see how I could put it in the body of the page (put `:::{version-dropdown}`... in the markdown), but I'm not sure how I would get that directive to render in the navbar. Is that possible?

**Comment by choldgraf** [comment]
Yep - you could use the theme part feature for the navbar! This is relatively new

https://myst-theme.netlify.app/ui/#navbar-end

**Comment by coretl** [comment]
I've got a first version of this ready here: 
https://github.com/DiamondLightSource/myst-version-switcher-plugin

The plugin code is here:
https://github.com/DiamondLightSource/myst-version-switcher-plugin/blob/main/plugins/version-switcher.mjs

There is also a GitHub action that assembles a `switcher.json` and GitHub pages site from release (and CI job) artifacts suitable for `actions/deploy-pages`

The docs are here (demonstrating usage):
https://diamondlightsource.github.io/myst-version-switcher-plugin/stable/

And I'm in the process of adopting it into some existing projects, e.g.:
https://pandablocks.github.io/PandABlocks-server/docs/
